conda/plugins/types.py

```diff
@@ -347,3 +349,31 @@ class CondaRequestHeader:
 
     name: str
     value: str
+
+
+@dataclass
+class CondaPostTransaction:
+    """
+    Return type to use when defining a post-transaction hook.
+
+    For details on how this is used, see
+    :meth:`~conda.plugins.hookspec.CondaSpecs.conda_post_transactions`.
+
+    :param name: Post transaction name (this is just a label).
+    :param run: Function to run after each transaction; accepts a single parameter which
+        is the action that has just been executed
+    :param action_type: Types of actions for which the hook should be run. Can either
+        be an Action subclass, or a tuple of Action subclasses
+    """
+
+    name: str
+    run: Callable[[Action], None]
+    action_type: type[Action] | tuple[type[Action]]
+
+    def __post_init__(self):
+        """Validate that the action_type is an Action subclass."""
+        if not is_only_subclass(self.action_type, Action):
```

Comment:
I don't think it makes sense to be doing this type of post-initialization type enforcement for our type classes. If we began doing it, it should be taken care of in a separate pull request and applied to all applicable type classes.

Let's just keep this pull request as simple as possible.